api/app/src/main/kotlin/packit/config/JwtLoginConfig.kt

```diff
@@ -0,0 +1,22 @@
+package packit.config
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.convert.DurationUnit
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+
+data class JwtPolicy(
+  val jwkSetURI: String,
+  val issuer: String,
+  val requiredClaims: Map<String, String> = mapOf(),
+  val grantedPermissions: List<String> = listOf(),
+
+  @DurationUnit(ChronoUnit.SECONDS)
+  val tokenDuration: Duration? = null,
+)
+
+@ConfigurationProperties(prefix = "auth.external-jwt")
```

Comment:
> do we need to add these into the applcicaton.properties

Because these are much more structured, I don't think it makes sense to use flat string environment variables to pass them in, but instead need to pass them as structured data.

There's a number of ways we can do it, see: https://docs.spring.io/spring-boot/reference/features/external-config.html

Reply:
> as per docs? or does it hyphen the camelcase?

So Spring is kind of weird and I got errors here if I didn't use kebab case here. It seems to apply some normalization to the input properties, which means that it doesn't actually matter what case people use, and both `external-jwt` and `externalJwt` would work.

See https://github.com/spring-projects/spring-boot/wiki/Canonical-properties
And https://github.com/spring-projects/spring-boot/blob/eb7b6a776dd8b01849a0ea35d097b81f7eeec23e/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/InvalidConfigurationPropertyNameFailureAnalyzer.java#L51-L52

Although you are right the docs I've written are inconsistent and use both. I'll switch to just one.

Reply:
> I don't think it makes sense to use flat string environment variables to pass them in, but instead need to pass them as structured data.

so yeah I thought this structured data is picked up from the application.properties? because how does it know what to set the values to ? i thought you need what you have in your docs as well?

Reply:
It does get populated from the properties in the `Environment`, but these can come from any number of sources, not just the built-in`application.properties`.

See https://docs.spring.io/spring-boot/reference/features/external-config.html

The simplest way of doing it is probably through the command line arguments.